Release notes 1.6.1 (#329)
NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver: release_notes/nrf_802154_1.6.1.md

# nRF IEEE 802.15.4 radio driver 1.6.1 release notes

Release date: December 9, 2019

This is a bugfix release that fixes problems found in the version 1.6.0.

> ## Bug fixes:

Reviewer: ```suggestion
## Bug fixes
```

> * Fix a problem when the CPU is halted during frame RX and FEM is disabled. ([PR #326](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/326))

Reviewer 2: How about `frame reception` instead of `frame RX`?

> * Fix a problem when the CPU is halted during frame RX and FEM is disabled. ([PR #326](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/326))

Reviewer: ```suggestion
* Fixed an issue where the CPU would be halted during the frame RX and the FEM would be disabled. ([PR #326](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/326))
```

> * Clear shared PPI channel when the driver stops using it. ([PR #328](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/328))

Reviewer: ```suggestion
* Fixed an issue where a PPI channel would not be cleared when the driver stopped using it. ([PR #328](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/328))
```

## Limitations
* The driver is tested only with the default configuration (that is, without changes in the nrf_802154_config.h file). Test applications are compiled with optimizations for the code size. Any other configuration and optimization level is to be considered as experimental.
* Number of supported peers for the Auto Pending Bit feature is limited to 127.
* In the multiprotocol configuration (SoftDevice RAAL), the application cannot use IRQ priority levels 2 and 3.

> * The delayed operations do not detect delays caused by the ISR latency. Because of the ISR latency, the delayed transmission might be delayed for too long, or the reception window might be shorter than requested for the delayed transmission operation. To avoid such problems, it is recommended to minimize the ISR latency of the RTC instance that is used by the lp_timer module (RTC2 by default).

Reviewer 2: Does it make sense to mention the default RTC instance if it depends on the chip used? (nRF52811 vs nRF52840)

# nRF IEEE 802.15.4 radio driver 1.6.1 release notes

Release date: December 9, 2019

This is a bugfix release that fixes problems found in the version 1.6.0.

## Bug fixes
* Fixed a spelling issue in the FEM GPIO macro. ([PR #286](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/286))
* Fixed an issue where building with GCC4 compiler was not possible. ([PR #298](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/298))
* Fixed an issue where the CPU would be halted during the frame reception and the FEM would be disabled. ([PR #326](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/326))
* Fixed an issue where a shared PPI channel would not be cleared when the driver stopped using it. ([PR #328](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/328))

> * Fixed an issue where PPI and TIMER HAL types were not casted explicitly. ([PR #330](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/330))

Reviewer 3: `were not cast explicitly`?

Reviewer: Correct, use `cast` here. Not though that `casted` is gaining ground again (again, as it was used in XVI century), but `cast` is the correct form here.

* Fixed an issue where building with GCC4 compiler was not possible. ([PR #298](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/298))
* Fixed an issue where the CPU would be halted during the frame reception and the FEM would be disabled. ([PR #326](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/326))
* Fixed an issue where a shared PPI channel would not be cleared when the driver stopped using it. ([PR #328](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/328))
* Fixed an issue where PPI and TIMER HAL types were not casted explicitly. ([PR #330](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/330))

## Limitations
* The driver is tested only with the default configuration (that is, without changes in the nrf_802154_config.h file). Test applications are compiled with optimizations for the code size. Any other configuration and optimization level is to be considered as experimental.
* Number of supported peers for the Auto Pending Bit feature is limited to 127.
* In the multiprotocol configuration (SoftDevice RAAL), the application cannot use IRQ priority levels 2 and 3.

> * The delayed operations do not detect delays caused by the ISR latency. Because of the ISR latency, the delayed transmission might be delayed for too long, or the reception window might be shorter than requested for the delayed transmission operation. To avoid such problems, it is recommended to minimize the ISR latency of the RTC instance that is used by the lp_timer module (RTC0 in NRF52811, otherwise RTC2).

Reviewer 4: I think `(RTC0 in NRF52811, otherwise RTC2)` is redundant here.
